tools/release-checklist

```diff
@@ -0,0 +1,326 @@
+make coreboot -j5
+make coreboot -j5 # must say "Fixpoint reached, bootstrap succeeded."
+git commit -m "change VERSION for $VERSION" -a
+git tag -m "release $VERSION" $VERSION
+
+# for actual releases, change VERSION into (N+1)+dev0; for example,
+#   4.08.0 => 4.08.1+dev0
+# for release candidates, use N+dev(D+1) instead; for example,
+#   4.07.0+rc2 => 4.07.0+dev10-2018-06-26
+git commit -m "increment version number after tagging $VERSION" VERSION
+git push
+git push --tags
+```
+
+## 6: create OPAM switches
+
+Create OPAM switches for the new version, copying the particular
+switch configuration choices from the previous version.
+
+I use a semi-automated process like the following:
+
+```
+# copy foo+rc1+... switches into foo+rc2+...
+for f in *+rc1*; do mkdir -p $f $(echo $f | sed 's/+rc1/+rc2/g'); done
+for f in *+rc1*/*; do cp $f $(echo $f | sed 's/+rc1/+rc2/g'); done
+git add *+rc2* # import into git (the files still use +rc1 all over)
+
+sed -i "s/+rc1/+rc2/g" *rc2*/*
+git diff # inspect the result
+git add *+rc2*
+
+sed -i "s/First release candidate/Second release candidate/g" *+rc2*/*
+git diff # inspect the result
+git add *+rc2*
+
+git commit -m "OPAM switches for $VERSION"
+```
+
+## 7: build the release archives
+
+```
+TMPDIR=/tmp/ocaml-release
+git checkout $VERSION
+git checkout-index -a -f --prefix=$TMPDIR/ocaml-$VERSION/
+cd $TMPDIR
+(cd ocaml-$VERSION && rm -rf experimental)
+gtar -c --owner 0 --group 0 -f ocaml-$VERSION.tar ocaml-$VERSION
+time gzip -9 <ocaml-$VERSION.tar >ocaml-$VERSION.tar.gz
+time xz <ocaml-$VERSION.tar >ocaml-$VERSION.tar.xz
+```
+
+## 8: upload the archive and compute checksums
+
+```
+scp ocaml-$VERSION.tar.{xz,gz} <<archive machine>>:$DIST
+```
+
+To update the checksum files on the remote host, we first upload the
+release environment as a script.
+
+
+```
+rm /tmp/env.sh
+echo export MAJOR=$MAJOR >> /tmp/env.sh
+echo export MINOR=$MINOR >> /tmp/env.sh
+echo export BUGFIX=$BUGFIX >> /tmp/env.sh
+echo export PLUSEXT=$PLUSEXT >> /tmp/env.sh
+
+echo export BRANCH=$BRANCH >> /tmp/env.sh
+echo export VERSION=$VERSION >> /tmp/env.sh
+
+echo export REPO=$REPO >> /tmp/env.sh
+echo export DIST=$DIST >> /tmp/env.sh
+
+# check env.sh
+cat /tmp/env.sh
+
+scp /tmp/env.sh <<archive machine>>:/tmp
+```
+
+and then login there to update the checksums (MD5SUM, SHA512SUM)
+
+```
+ssh <<archive machine>>
+source /tmp/env.sh
+cd $DIST
+
+cp MD5SUM MD5SUM.old
+md5sum ocaml-$VERSION.tar.{xz,gz} > new-md5s
+# check new-md5s to ensure that they look right, and then
+cat new-md5s >> MD5SUM
+# if everything worked well,
+rm MD5SUM.old
+
+# same thing for SHA512
+cp SHA512 SHA512.old
+cp SHA512SUM SHA512SUM.old
+sha512sum ocaml-$VERSION.tar.{xz,gz} > new-sha512s
+cat new-sha512s >> SHA512SUM
+rm SHA512SUM.old
+```
+
+## 9: update note files (technical documentation)
+
+```
+ssh <<archive machine>> "mkdir -p $DIST/notes"
+scp INSTALL.adoc LICENSE README.adoc README.win32.adoc Changes <<archive machine>>:$DIST/notes/
```

Comment:
This line is over-long (though perhaps that just means `ocaml-typo=long-line` is wanted in `.gitattributes` for this file)